src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -62,6 +62,66 @@ size_t AdaptDispatch::_SetRgbColorsHelper(const VTParameters options,
     return optionsConsumed;
 }
 
+// Routine Description:
+// - Helper to parse extended graphics options, which start with 38 (FG) or 48 (BG)
+//   - These options are followed by either a 2 (RGB) or 5 (xterm index):
+//     - RGB sequences then take 4 MORE options to designate the ColorSpaceID, R, G, B parts
+//       of the color.
+//     - Xterm index will use the option that follows to use a color from the
+//       preset 256 color xterm color table.
+// Arguments:
+// - colorItem - One of FG(38) and BG(48), indicating which color we're setting.
+// - options - An array of options that will be used to generate the RGB color
+// - attr - The attribute that will be updated with the parsed color.
+// Return Value:
+// - <none>
+void AdaptDispatch::_SetRgbColorsHelperAlt(const VTParameter colorItem,
+                                           const VTSubParameters options,
+                                           TextAttribute& attr) noexcept
+{
+    // This should be called for applying FG and BG colors only.
+    assert(colorItem == GraphicsOptions::ForegroundExtended ||
+           colorItem == GraphicsOptions::BackgroundExtended);
+
+    const bool isForeground = (colorItem == GraphicsOptions::ForegroundExtended);
+    const DispatchTypes::GraphicsOptions typeOpt = options.at(0);
+
+    if (typeOpt == DispatchTypes::GraphicsOptions::RGBColorOrFaint)
+    {
+        // sub params are in the order:
+        // :2:<color-space-id>:<r>:<g>:<b>
+        // we skip color-space-id, because we don't support it.

```

Comment:
I'm wondering whether maybe we should check the color space parameter here, and if it's anything other than "default", then we just don't apply that color.

The reason why I say that is that some terminals initially supported these sequences _without_ the color space parameter (in fact some probably still do). So if we receive a sequence like `2:12:34:56`, it's more likely to be an attempt to set RGB(12,34,56) rather than RGB(34,56,0).

If you're feeling very enthusiastic, we could actually support both forms (so both `2:12:34:56` and `2::12:34:56` produce the same result), but I'd be quite happy to just ignore anything with an "invalid" color space, and hope that discourages apps from using that format. I just don't want us to interpret those sequences as the _wrong_ color.

Reply:
I think we should just treat this form `2:12:34:56` as invalid. It feels better to start with right semantics when we can as we are already trapped by semi-colons 😄

This shouldn't be difficult to implement either, just need to check `options.at(1).has_value()`.